### PR TITLE
ome demoserver to 5.4.1

### PR DESCRIPTION
**Pull request by kennethgillen**
Upgrading ome-demoserver to OMERO 5.4.1

(Sibling PR: https://github.com/openmicroscopy/management_tools/pull/610/)

### CODE REVIEW

File: requirements.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 - name: openmicroscopy.omero-server
   src:  https://github.com/openmicroscopy/ansible-role-omero-server.git
-  version: 2.0.0-m1
+  version: 2.0.1
```
Comment (sbesson) [inline requirements.yml]: Now `2.0.2` is out, should this be bumped as I understand the playbook was deployed with the role changes integrated in https://github.com/openmicroscopy/ansible-role-omero-server/pull/22?

**Comment by kennethgillen** [comment]
The playbook at https://github.com/openmicroscopy/prod-playbooks/pull/17/commits/0b80f1912b36bb6bb121155c1fff67aecf69f2c6 upgraded ome-demoserver to 5.4.1 with a modified version of 2.0.1 of ansible-role-omero-server matching 2.0.2 https://galaxy.ansible.com/openmicroscopy/omero-server/

File: ome-demoserver.yml [hunk 2]
```diff
@@ -126,10 +126,13 @@
       omero_server_dbpassword: "{{ vault.omero_server_dbpassword }}"
       omero_server_rootpassword: "{{ vault.omero_server_rootpassword }}"
       omero_server_systemd_limit_nofile: 16384
+      # Once omego version bug is fixed, we can remove this hard-coding.
```
Comment (joshmoore) [inline ome-demoserver.yml]: Perhaps add a `TODO`

Comment (kennethgillen) [reply ome-demoserver.yml]: Do you mean something more than just adding TODO do the comment? (a la --Fixes in github?) e.g. something for SCC?

Comment (joshmoore) [reply ome-demoserver.yml]: Perfect. Thanks.

**Comment by kennethgillen** [comment]
Is it worth figuring out the syntax for https://github.com/openmicroscopy/ansible-role-postgresql/tree/3.0.1 in this PR?

It would mean working out how to use the new variables (`postgresql_users_databases` was split).

I don't see a reason to bump it, other than to keep up to date with the roles / or test the roles on our production infrastructure.

File: requirements.yml [hunk 3]
```diff
@@ -16,7 +16,7 @@
 
 - name: openmicroscopy.omero-server
   src:  https://github.com/openmicroscopy/ansible-role-omero-server.git
```
Comment (kennethgillen) [inline requirements.yml]: Should these all come from Galaxy, rather than GH now?

Comment (manics) [reply requirements.yml]: Yes

Comment (joshmoore) [reply requirements.yml]: Are you pushing this, @kennethgillen ?

Comment (kennethgillen) [reply requirements.yml]: Yes, will pull the GH urls. (Does it apply to all requirements?)

Comment (kennethgillen) [reply requirements.yml]: Ah, @manics already replied to my first question which was "all". Pushing.

**Comment by kennethgillen** [comment]
```
 [WARNING]: - omero-user was NOT installed successfully: Role has no field
named u'owner'
```

**Comment by kennethgillen** [comment]
@manics  - one "switch to galaxy" too far?

**Comment by joshmoore** [comment]
Missing `openmicroscopy.` prefix.

**Comment by kennethgillen** [comment]
d'oh. Thanks.

**Comment by manics** [comment]
Can you match the syntax in the rest of the requirements file: `src:` instead of `name:`?

**Comment by kennethgillen** [comment]
Do you/we prefer `src` or `name`?

```
LS27172:prod-playbooks kenny$ cat requirements.yml  | grep -E "src|name" | awk '{print $2}' | sort | uniq -c
   7 name:
   3 src:
```

**Comment by sbesson** [comment]
From http://docs.ansible.com/ansible/latest/galaxy.html#installing-multiple-roles-from-a-file, `src` is required and `name` is optional. Since we should have fixed the Galaxy names, I am in favor of unifying towards `src: openmicroscopy.<role>` whenever possible.

Re style guide, :+1: I'm slowly going through our contributing documentation and Ansible is definitely one of my upcoming languages if no-one wants to take a first stab.

**Comment by sbesson** [comment]
Thanks. With the last changes, I think we brought these playbooks to the latest known state of the Ansible roles. If no objection from @joshmoore, I would propose we get this and the associated inventory PR in and mark the completton of the prod servers upgrade to 5.4.1 so that we can focus on the next tasks.

**Comment by joshmoore** [comment]
:+1: for getting this in as is. I lost track a bit of the `omero-marshal` discussion, so not sure if anything needs to be captured from that discussion.

**Comment by sbesson** [comment]
My understanding is that at the level of our prod OMERO playbooks, we are currently leaving `omero-marshal`  fully managed alongside the other OMERO.web dependencies via the `requirements-py27.txt` file shipped with the bundle. Concretely this means upgrades will be triggered by version bump in the upstream code - see https://github.com/openmicroscopy/openmicroscopy/pull/5595.

We can re-evaluate this strategy when we have a need for a redeployment of the Web client with a  a new version of `omero-marshal`.

**Comment by kennethgillen** [comment]
@sbesson - there are some newer roles such as the PostgreSQL role, but that'll take either input from @manics or some testing time by me to work how how to set the variables to end up with the configuration we want.

**Comment by sbesson** [comment]
Understood. Including these as they are then.

**Pull request merged**